framework/include/cppmicroservices/SharedLibrary.h

```diff
@@ -35,186 +37,200 @@
-   * Sets the file name prefix for shared library names (e.g. .dll or .so).
-   * Does nothing if the shared library is already loaded or the
-   * SharedLibrary(const std::string&) constructor was used.
-   *
-   * @param prefix The shared library name prefix.
-   */
-  void SetPrefix(const std::string& prefix);
-
-  /**
-   * Gets the file name prefix of this SharedLibrary object.
-   *
-   * @return The file name prefix of the shared library.
-   */
-  std::string GetPrefix() const;
-
-  /**
-   * Gets the internal handle of this SharedLibrary object.
-   *
-   * @return \c nullptr if the shared library is not loaded, the operating
-   * system specific handle otherwise.
-   */
-  void* GetHandle() const;
-
-  /**
-   * Gets the loaded/unloaded stated of this SharedLibrary object.
-   *
-   * @return \c true if the shared library is loaded, \c false otherwise.
-   */
-  bool IsLoaded() const;
-
-private:
-  ExplicitlySharedDataPointer<SharedLibraryPrivate> d;
-};
+  class US_Framework_EXPORT SharedLibrary
+  {
+    public:
+    SharedLibrary();
+    SharedLibrary(const SharedLibrary& other);
+
+    /**
+     * Construct a SharedLibrary object using a library search path and
+     * a library base name.
+     *
+     * @param libPath An absolute path containing the shared library
+     * @param name The base name of the shared library, without prefix
+     *        and suffix.
+     */
+    SharedLibrary(const std::string& libPath, const std::string& name);
+
+    /**
+     * Construct a SharedLibrary object using an absolute file path to
+     * the shared library. Using this constructor effectively disables
+     * all setters except SetFilePath().
+     *
+     * @param absoluteFilePath The absolute path to the shared library.
+     */
+    SharedLibrary(const std::string& absoluteFilePath);
+
+    /**
+     * Destroys this object but does not unload the shared library.
+     */
+    ~SharedLibrary();
+
+    SharedLibrary& operator=(const SharedLibrary& other);
+
+    /**
+     * Loads the shared library pointed to by this SharedLibrary object.
+     * On POSIX systems dlopen() is called with the RTLD_LAZY and
+     * RTLD_LOCAL flags unless the compiler is gcc 4.4.x or older. Then
+     * the RTLD_LAZY and RTLD_GLOBAL flags are used to load the shared library
+     * to work around RTTI problems across shared library boundaries.
+     *
+     * @throws std::logic_error If the library is already loaded.
+     * @throws std::runtime_error If loading the library failed.
+     */
+    void Load();
+    
+    /**
+     * Loads the shared library pointed to by this SharedLibrary object.  On
+     * POSIX systems dlopen() is called with flags from the key
+     * LIBRARY_LOAD_OPTIONS in the frameworkProperties map.
+     *
+     * @throws std::logic_error If the library is already loaded.
+     * @throws std::runtime_error If loading the library failed.
+     */
+    void Load(const std::unordered_map<std::string, Any>& frameworkProperties);
```

Comment:
I'm not sure that another ```Load``` function is necessary. It seems like it only adds complexity. Since a client sets the ```LIBRARY_LOAD_OPTIONS``` config option using the actual POSIX ```RTLD_*``` values, they can be passed directly to ```Load(int flags)```.